packages/amplify-category-geo/src/service-utils/deviceLocationTrackingUtils.ts

```diff
@@ -0,0 +1,165 @@
+import { App } from '@aws-cdk/core';
+import {
+  $TSContext, $TSObject, pathManager, stateManager,
+} from 'amplify-cli-core';
+import { printer } from 'amplify-prompts';
+import { category } from '../constants';
+import { getTemplateMappings } from '../provider-controllers';
+import { DeviceLocationTrackingStack } from '../service-stacks/deviceLocationTrackingStack';
+import { parametersFileName, provider, ServiceName } from './constants';
+import { DeviceLocationTrackingParameters } from './deviceLocationTrackingParams';
+import {
+  generateTemplateFile, getAuthResourceName, getResourceDependencies, readResourceMetaParameters, ResourceDependsOn, updateDefaultResource, updateParametersFile,
+} from './resourceUtils';
+
+/**
+ * createDeviceLocationTrackingResource
+ */
+export const createDeviceLocationTrackingResource = async (
+  context: $TSContext,
+  parameters: DeviceLocationTrackingParameters,
+): Promise<void> => {
+  const authResourceName = await getAuthResourceName(context);
+  // generate CFN files
+  const templateMappings = await getTemplateMappings(context);
+
+  const mapStack = new DeviceLocationTrackingStack(
+    new App(),
+    'DeviceLocationTrackingStack',
+    { ...parameters, ...templateMappings, authResourceName },
+  );
+  generateTemplateFile(mapStack, parameters.name);
+  saveCFNParameters(parameters);
+  stateManager.setResourceInputsJson(
+    pathManager.findProjectRoot(),
```

Comment:
When the first arg is undefined, stateManager will call pathManager.findProjectRoot()

```suggestion
    undefined,
```